src/org/pentaho/reporting/platform/plugin/cache/DeleteOldOnAccessCache.java

```diff
@@ -81,18 +92,32 @@ private void cleanUp() {
     final long currentTimeMillis = System.currentTimeMillis();
     final ICacheBackend backend = getBackend();
     //Check all timestamps
-    for ( final String key : backend.listKeys( Collections.singletonList( SEGMENT ) ) ) {
-      if ( key.matches( ".*" + TIMESTAMP ) ) {
-        final List<String> compositeKey = Arrays.asList( SEGMENT, key );
-        final Long timestamp = (Long) backend.read( compositeKey );
-        if ( currentTimeMillis - timestamp > millisToLive ) {
-          backend.purge( compositeKey );
-          compositeKey.set( 1, compositeKey.get( 1 ).replace( TIMESTAMP, "" ) );
-          backend.purge( compositeKey );
-          logger.debug( "Purged long-term cache: " + key );
+    for ( final String userFolder : backend.listKeys( Collections.singletonList( SEGMENT ) ) ) {
+      for ( final String key : backend.listKeys( Arrays.asList( SEGMENT, userFolder ) ) ) {
+        if ( key.matches( ".*" + TIMESTAMP ) ) {
+          final List<String> compositeKey = Arrays.asList( SEGMENT, userFolder, key );
+          final Long timestamp = (Long) backend.read( compositeKey );
+          if ( currentTimeMillis - timestamp > millisToLive ) {
+            backend.purge( compositeKey );
+            compositeKey.set( NAME_INDEX, compositeKey.get( NAME_INDEX ).replace( TIMESTAMP, "" ) );
+            backend.purge( compositeKey );
+            logger.debug( "Purged long-term cache: " + key );
+          }
         }
       }
     }
     logger.debug( "Finished periodical cache eviction" );
   }
+
+
```

Comment:
Check key == null, or risk crashes. Session.getName() can be null if the user is not authenticated (ie you are anonymous).